Source/Core/VideoBackends/Software/SWVertexLoader.h

```diff
@@ -14,11 +15,37 @@
 
 #include "VideoCommon/VertexManagerBase.h"
 
+// There is somewhat odd behavior where uninitialized components use values from 16 vertices ago
+// (only counting vertices that had initialized data).
+template <typename T>
+struct CacheEntry
+{
+  static constexpr auto CACHE_SIZE = 16;
+
+  std::array<T, CACHE_SIZE> data;
+  u8 index;
+
+  T Read() { return data[index]; }
+  void Write(const T& value)
+  {
+    data[index] = value;
+    index = (index + 1) % CACHE_SIZE;
+  }
+};
```

Comment:
I've been calling this a cache, but I don't know if that's the best description.  It's definitely unrelated to the vertex cache, which [libogc claims](https://github.com/devkitPro/libogc/blob/cb1934ccfde50373236afec894c21bb620dfe9ac/gc/ogc/gx.h#L1764-L1766) is not used for direct components.

Perhaps calling it a queue or pipeline would be better, along the lines of "each vertex component has a 16-element queue for values, written into when that component is enabled.  If the component is not enabled, the value in the index that would have been written to next (i.e. the value from 16 components ago) is used, and the index does not change".

I also looked at the various patents to try and understand things better; the closest I've found is "Data required to process a particular component is stored within a queue 410 having a depth that is proportional to memory latency." in [US6717577B1](https://patents.google.com/patent/US6717577B1/en) (see Fig. 4 / image 9), but there's no guarantee that exactly describes what is used in practice.